### PR TITLE
Contact crud

### LINKED ISSUE #79: CRUD + Schema validation for contact
CRUD + Schema validation for contact. See Wiki for details.

Endpoints under Contact controller:
- GET /api/contact/{contact_id}/
- UPDATE /api/contact/{contact_id}/
- DELETE /api/contact/{contact_id}/

**Note:** Below is kind of weird. Not certain if best way.

Endpoints under Employer controller (Because you only ever make a contact that is for an employer):
- POST /api/employer/{employer_id}/contact

### PR DESCRIPTION
**Pull request by ganeshasapu**
# Description
Please check out the createEmployer method in the employer.ts file. I did some stuff to create contactIDs which I am not 100% sure is correct, so please let me know if there is a better way to do this!

Fixes #79, #77
* Creates CRUD for contact object
* Makes quite a lot of changes for employer object
* Adds schema validation for both contact and employer
___
### Employer - POST
curl --location --request POST 'http://localhost:3001/api/employer' \
--header 'Content-Type: application/json' \
--data-raw '{
	"name": "Test",
	"primary_address_id": "63bb1f16939f28affbacf5fd",
	"contacts": [
		{
            "phone_num": "42423", 
            "email": "434234"
        }
	],
	"timeline": "63bb1f16939f28affbacf5fb",
	"creator_id": "63bb1f16939f28affbacf5fc",
	"owner_id": "63bb1f16939f28affbacf5fa",
	"legal_name": null,
	"hr_contacts": null,
	"secondary_address_id": null,
	"initial_contact": null
}'

### Response
Created an employer with id: 63d9d1a9d02951f4ff42430b
___
### Employer - GET
curl --location --request GET 'http://localhost:3001/api/employer/63d9d34059e2998c662a24ae'

### Response
{
    "_id": "63d9d34059e2998c662a24ae",
    "name": "Test",
    "primary_address_id": "63bb1f16939f28affbacf5fd",
    "contacts": [
        "63d9d34059e2998c662a24ad"
    ],
    "creator_id": "63bb1f16939f28affbacf5fc",
    "owner_id": "63bb1f16939f28affbacf5fa",
    "timeline": "63bb1f16939f28affbacf5fb",
    "legal_name": null,
    "hr_contacts": null,
    "secondary_address_id": null,
    "initial_contact": null
}
___
### Contact - POST
curl --location --request POST 'http://localhost:3001/api/employer/63d9d1a9d02951f4ff42430b/contact' \
--header 'Content-Type: application/json' \
--data-raw '{
    "phone_num": "42423", 
    "email": "434234"
}'

* **Note that 63d9d1a9d02951f4ff42430b is ID of an employer**
*  **the contacts array of that employer has been updated with the created contact's ID (This can be verified using Employer - GET)**

### Response
Successfully created a new contact with id 63d9d3e259e2998c662a24b0
___
### Contact - GET
curl --location --request GET 'http://localhost:3001/api/contact/63d9d3e259e2998c662a24b0'

### Response
{
    "_id": "63d9d3e259e2998c662a24b0",
    "phone_num": "42423",
    "email": "434234",
    "alt_phone_num": null,
    "website": null
}
___
### Contact - DELETE
curl --location --request DELETE 'http://localhost:3001/api/contact/63d9d3e259e2998c662a24b0'

### Response
Successfully removed contact with id 63d9d3e259e2998c662a24b0
___
### Contact - PUT
curl --location --request PUT 'http://localhost:3001/api/contact/63d9d45a59e2998c662a24b1' \
--header 'Content-Type: application/json' \
--data-raw '{
    "phone_num": "11111", 
    "email": "4341111111234"
}'

### Response
Successfully updated contact with id 63d9d45a59e2998c662a24b1
___
## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] I have run `npm run lint:fix`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works (Delete if frontend)

### CODE REVIEW

**Review by helen-li — COMMENTED** [review]
Slayed overall - I didn't run the branch locally on my setup but looked through the code closely and all made sense to me - just left 2 really minor things

**Review by jordanjanakievski — APPROVED** [review]
I took a quick look at the code, but I am going to leave that to you and Helen for those styling details and specifics

I checked all the requests and everything worked smoothly.

**Review by helen-li — APPROVED** [review]
Yuh

File: server/src/controllers/contact.controller.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  next: NextFunction
+) => {
+  try {
+    const id = req?.params?.id;
+    const result = await Contact.updateContact(id, req.body);
+    res.status(200).send(result);
+  } catch (e) {
+    if (e instanceof Error) {
+      error(e.message);
+      res.status(304).send(e.message);
```
Comment (kenneth-miura) [inline server/src/controllers/contact.controller.ts:63]: why is this a 304

Comment (kenneth-miura) [reply server/src/controllers/contact.controller.ts:63]: should be 404 no?

**Review by kenneth-miura — COMMENTED** [review]
i think http response code is wrong?

**Pull request merged**